include/fmt/core.h

```diff
@@ -672,20 +673,21 @@ class basic_parse_context : private ErrorHandler {
 
   explicit FMT_CONSTEXPR basic_parse_context(
       basic_string_view<Char> format_str, ErrorHandler eh = ErrorHandler())
-    : ErrorHandler(eh), format_str_(format_str), next_arg_id_(0) {}
+    : ErrorHandler(eh), begin_(format_str.begin()), end_(format_str.end()),
+      next_arg_id_(0) {}
 
   // Returns an iterator to the beginning of the format string range being
   // parsed.
   FMT_CONSTEXPR iterator begin() const FMT_NOEXCEPT {
-    return format_str_.begin();
+    return begin_;
   }
 
   // Returns an iterator past the end of the format string range being parsed.
-  FMT_CONSTEXPR iterator end() const FMT_NOEXCEPT { return format_str_.end(); }
+  FMT_CONSTEXPR iterator end() const FMT_NOEXCEPT { return end_; }
 
   // Advances the begin iterator to ``it``.
   FMT_CONSTEXPR void advance_to(iterator it) {
-    format_str_.remove_prefix(it - begin());
```

Comment:
VS17 asserts that iterators must point to the same string_views:
```C++
constexpr difference_type operator-(const _String_view_iterator& _Right) const _NOEXCEPT
{	// return difference of iterators
#if _ITERATOR_DEBUG_LEVEL >= 1
	_IDL_VERIFY(_Mydata == _Right._Mydata && _Mysize == _Right._Mysize,
		"cannot subtract incompatible string_view iterators");
```
After remove_prefix both `format_str_`'s data and size will be different and this check won't pass.
I've replaced format_str_ with a pair of string_view iterators to resolve this.